src/marker.rs

```diff
@@ -0,0 +1,85 @@
+//! This module contains marker traits used by `wasm-bindgen` to verify its
+//! generated code.
+//!
+//! NOTE: This module is not exported as is. It is re-exported under
+//! `wasm_bindgen::__rt::marker`.
+//!
+//! # ⚠️ Unstable
+//!
+//! This is an internal module, no stability guarantees are provided. Use at
+//! your own risk.
+
+#[cfg(feature = "diagnostic")]
+extern crate rustversion;
+
+/// Marker trait for types that support `#[wasm_bindgen(constructor)]`.
+///
+/// **DO NOT** implement this trait manually. It is implemented automatically
+/// for types that support constructors.
+///
+/// # ⚠️ Unstable
+///
+/// This is part of the internal [`convert`](crate::marker) module, **no
+/// stability guarantees** are provided. Use at your own risk. See its
+/// documentation for more details.
+#[cfg_attr(
+    feature = "diagnostic",
+    rustversion::attr(
+        since(1.78),
+        diagnostic::on_unimplemented(
+            message = "JavaScript constructors are not supported for `{Self}`",
+            label = "this function cannot be the constructor of `{Self}`",
+            note = "`#[wasm_bindgen(constructor)]` is generally only supported for `struct`s with `#[wasm_bindgen]` and cannot be used for `enum`s.",
```

Comment:
```suggestion
            note = "`#[wasm_bindgen(constructor)]` is only supported for `struct`s and cannot be used for `enum`s.",
```
Applies to the others as well.